src/OmniSharp.Abstractions/Utilities/Platform.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.IO;
+
+namespace OmniSharp.Utilities
+{
+    public enum OperatingSystem
+    {
+        Unknown,
+        Windows,
+        MacOS,
+        Linux
+    }
+
+    public enum Architecture
+    {
+        Unknown,
+        x86,
+        x64
+    }
+
+    public sealed class Platform
+    {
+        public static Platform Current { get; } = GetCurrentPlatform();
+
+        public OperatingSystem OperatingSystem { get; }
+        public Architecture Architecture { get; }
+        public Version Version { get; }
+        public string LinuxDistributionName { get; }
+
+        private Platform(
+            OperatingSystem os = OperatingSystem.Unknown,
+            Architecture architecture = Architecture.Unknown,
+            Version version = null,
+            string linuxDistributionName = null)
+        {
+            OperatingSystem = os;
+            Architecture = architecture;
+            Version = version ?? new Version(0, 0);
+            LinuxDistributionName = linuxDistributionName ?? string.Empty;
+        }
+
+        private static Platform GetCurrentPlatform()
+        {
+            var os = OperatingSystem.Unknown;
+            var architecture = Architecture.Unknown;
+
+            // Simple check to see if this is Windows
+            var platformId = (int)Environment.OSVersion.Platform;
+            if (platformId <= 3 || platformId == 5)
```

Comment:
Can you add a comment explaining where this knowledge came from?

Reply:
I'll try to remember where it came from. This is derived from `scripts/platform.cake`

Reply:
Oh right. It was [this](https://msdn.microsoft.com/en-us/library/3a8hyw88(v=vs.110).aspx):

| Member       | Underlying value |
|--------------|------------------|
| Win32S       | 0                |
| Win32Windows | 1                |
| Win32NT      | 2                |
| WinCE        | 3                |
| Unix         | 4                |
| Xbox         | 5                |
| MacOSX       | 6                |

Will add a comment